static/webmention.js

```diff
@@ -418,48 +393,62 @@ A more detailed example:
       apiURL += `&target[]=${encodeURIComponent('http:' + path)}&target[]=${encodeURIComponent('https:' + path)}`;
     });
 
-    getData(apiURL, function(/** @type {WebmentionResponse} */json) {
-      /** @type {Array<Reaction>} */
-      let comments = [];
-      /** @type {Array<Reaction>} */
-      const collects = [];
-
-      if (commentsAreReactions) {
-        comments = collects;
+    /** @type {WebmentionResponse} */
+    let json = {};
+    try {
+      const response = await window.fetch(apiURL);
+      if (response.status >= 200 && response.status < 300) {
+        json = await response.json();
+      } else {
+        console.error("Could not parse response");
+        new Error(response.statusText);
       }
+    } catch(error) {
+      console.error("Request failed", error);
+      // Burn the app with fire 🔥
+      throw error;
+    }
 
-      /** @type {Record<ReactEmoji, Array<Reaction>>} */
-      const mapping = {
-        "in-reply-to": comments,
-        "like-of": collects,
-        "repost-of": collects,
-        "bookmark-of": collects,
-        "follow-of": collects,
-        "mention-of": comments,
-        "rsvp": comments
-      };
-
-      json.children.forEach(function(child) {
-        // This seem to push the reaction into either comments or collects
-        const store = mapping[child['wm-property']];
-        if (store) {
-          store.push(child);
-        }
-      });
+    /** @type {Array<Reaction>} */
+    let comments = [];
+    /** @type {Array<Reaction>} */
+    const collects = [];
 
-      // format the comment-type things
-      let formattedComments = '';
-      if (comments.length > 0 && comments !== collects) {
-        formattedComments = formatComments(dedupe(comments));
-      }
+    if (commentsAreReactions) {
+      comments = collects;
+    }
 
-      // format the other reactions
-      let reactions = '';
-      if (collects.length > 0) {
-        reactions = formatReactions(dedupe(collects));
+    /** @type {Record<ReactEmoji, Array<Reaction>>} */
+    const mapping = {
+      "in-reply-to": comments,
+      "like-of": collects,
+      "repost-of": collects,
+      "bookmark-of": collects,
+      "follow-of": collects,
+      "mention-of": comments,
+      "rsvp": comments
+    };
+
+    json.children.forEach(function(child) {
+      // This seem to push the reaction into either comments or collects
```

Comment:
Make this comment "put the reaction into its appropriate destination"

Reply:
I've added that comment, because I wasn't quite sure what's going on there.
Feels a bit dirty to have the `store` point to the respective object in memory …
=> Another issue?

Reply:
There is nothing dirty about that pattern. It's simple and straightforward and efficient.